src/colorizer/state/ViewerState.ts

```diff
@@ -47,7 +50,7 @@ export const viewerStateStoreCreator: StateCreator<ViewerState> = (...a) => ({
  * console.log(store.dataset);
  * ```
  */
-export const useViewerStateStore = create<ViewerState>()(viewerStateStoreCreator);
+export const useViewerStateStore = create<ViewerState>()(subscribeWithSelector(viewerStateStoreCreator));
```

Comment:
This allows subscribers to changes in the store to add a selector function to select a subset of state. This allows those subscribers to only be called when those dependencies change.